pkg/ingester/ingester_v2.go

```diff
@@ -677,6 +681,12 @@ func (i *Ingester) updateLoop(ctx context.Context) error {
 			i.ingestionRate.Tick()
 		case <-rateUpdateTicker.C:
 			i.userStatesMtx.RLock()
+
+			if i.TSDBState.closed {
```

Comment:
This cannot possibly happen. `updateLoop` is called while Ingester is in `Running` state. Only when this function returns, ingester goes to `Stopping` state and will call `closeAllTSDB`.

Reply:
That makes sense, thanks for catching that mistake.